client/src/partials/depots/entry/entry.js

```diff
@@ -0,0 +1,334 @@
+      return drug.validLots;
+    });
+  }
+
+  function setSessionProperties (models) {
+    if (models.orders.data.length < 1) {
+      return $q.reject('ERROR.EMPTY_DATA');
+    }
+
+    // deactivate find
+    find.valid = true;
+    find.active = false;
+
+    // set up watchers for totalling and validation
+    $scope.$watch('session.order.data', calculateTotals, true);
+    $scope.$watch('session.order.data', validateSession, true);
+
+    session.order = models.orders;
+
+    // set up session properties
+    session.cfg.is_direct = models.orders.data[0].is_direct;
+    session.cfg.order_date = new Date(models.orders.data[0].purchase_date);
+
+    if(!session.cfg.is_direct){
+      session.cfg.employee_id = models.orders.data[0].purchaser_id;
+      session.cfg.employee_name = ($scope.employees.get(session.cfg.employee_id).prenom || '') + ' ' + ($scope.employees.get(session.cfg.employee_id).name || '');
+    }
+
+    // modify paramters
+    session.order.data.forEach(function (drug) {
+      drug.lots = new Store({ identifier : 'tracking_number', data : [] });
+      angular.extend(drug, { isCollapsed : false });
+      //drug.isCollapsed = false;
+      //drug.edittable = true;
+      $scope.addLot(drug);
+    });
+  }
+
+  function sum (a, b) {
+    return a + Number(b.quantity);
+  }
+
+  function calculateTotals () {
+    if (!session.order || !session.order.data) { return; }
+
+    // total and calculate metadata
+    var totals = session.totals;
+
+    totals.quantity = 0;
+    totals.price = 0;
+    totals.purchase_price = 0;
+    totals.items = session.order.data.length;
+    session.order.data.forEach(function (drug) {
+      totals.quantity += precision.round(drug.quantity);
+      totals.price += precision.round(drug.unit_price * drug.quantity);
+      drug.totalQuantity = drug.lots.data.reduce(sum, 0);
+      drug.validLots = valid(drug.lots) && drug.totalQuantity === drug.quantity;
+    });
+  }
+
+  function valid (lots) {
+    var isDef = angular.isDefined;
+    return lots.data.every(function (row) {
+      var newDate = new Date().getTime(),
+        expirate = new Date(row.expiration_date).getTime(),
+        diffDays = (parseInt((expirate-newDate)/(24*3600*1000)));
+
+      var n = parseFloat(row.quantity);
+      return n > 0 && (diffDays > 0) && isDef(row.lot_number) &&
+        isDef(row.expiration_date) &&
+        !!row.lot_number;
+    });
+  }
+
+  $scope.cancel = function cancel () {
+    session = $scope.session = { cfg : {}, totals : [] };
+    find.fn.reset();
+  };
+
+  $scope.expand = function expand (drug) {
+    drug.isCollapsed = !drug.isCollapsed;
+  };
+
+  function Lot () {
+    this.inventory_uuid = null;
+    this.purchase_order_uuid = null;
+    this.expiration_date = new Date();
+    this.date = new Date();
+    this.lot_number = null;
+    this.tracking_number = uuid();
+    this.quantity = 0;
+  }
+
+  $scope.addLot = function addLot (drug) {
```

Comment:
Because it would take to long to rewrite the module and test that it all works.  I'm just trying to change the links in this PR.